agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
-        mg.setInstructionList(il);
-        mg.update();
-
-        // Update the max stack and Max Locals
-        mg.setMaxLocals();
-        mg.setMaxStack();
-        mg.update();
-
-        // Update the method in the class
-        cg.replaceMethod(methods[i], mg.getMethod());
-        if (debug) out.format("Modified code: %s%n", mg.getMethod().getCode());
-
-        // verify the new method
-        // StackVer stackver = new StackVer();
-        // VerificationResult vr = stackver.do_stack_ver (mg);
-        // log ("vr for method %s = %s%n", mg.getName(), vr);
-        // if (vr.getStatus() != VerificationResult.VERIFIED_OK) {
-        // System.out.printf ("Warning BCEL Verify failed for method %s: %s",
-        // mg.getName(), vr);
-        // System.out.printf ("Code: %n%s%n", mg.getMethod().getCode());
-        // System.exit(1);
-        // }
+      if (transformMethod(mg, new InstructionFactory(cg), pgen)) {
+        transformed = true;
       }
 
-      cg.update();
-    } catch (Exception e) {
-      out.format("Unexpected exception encountered: " + e);
-      e.printStackTrace();
+      // Update the instruction list
+      mg.setInstructionList(instructionList);
+      mg.update();
+
+      // Update the max stack and Max Locals
+      mg.setMaxLocals();
+      mg.setMaxStack();
+      mg.update();
+
+      // Update the method in the class
+      cg.replaceMethod(method, mg.getMethod());
+
+      debug_instrument_inst.log(
+          "%s.%s modified code: %s%n%n", mg.getClassName(), mg.getName(), mg.getMethod().getCode());
     }
+    cg.update();
 
+    debug_transform.log("transformClass: EXIT %s", cg.getClassName());
     return transformed;
   }
 
   /**
-   * Instrument the specified method to replace mapped calls.
+   * Transforms the specified method to replace mapped calls.
    *
-   * @param m the method to transform
    * @param mg the method generator
-   * @param current_method_map the current map of method definitions
+   * @param ifact the instrument factory for the enclosing class of this method
+   * @param pgen the constant pool for this class
    */
-  public void instrument_method(
-      Method m, MethodGen mg, Map<MethodDef, ReplacementClass> current_method_map) {
-
+  private boolean transformMethod(MethodGen mg, InstructionFactory ifact, ConstantPoolGen pgen) {
+    boolean transformed = false;
     // Loop through each instruction, making substitutions
-    InstructionList il = mg.getInstructionList();
-    for (InstructionHandle ih = il.getStart(); ih != null; ) {
-      if (debug_instrument_inst.enabled()) {
-        debug_instrument_inst.log("instrumenting instruction %s%n", ih);
-        // ih.getInstruction().toString(pool.getConstantPool()));
+    debug_transform.log("transformMethod: ENTER %s.%s%n", mg.getClassName(), mg.getName());
+    InstructionList instructionList = mg.getInstructionList();
+    for (InstructionHandle instructionHandle = instructionList.getStart();
```

Comment:
This is a non-standard way to use a `for` loop.  I would make it a `while` loop, since that's what it is.
(The only advantage of using a `for` loop is to scope `instructionHandle` to the body, which is useful but not worth the difficulty of understanding it.)